### DISCUSSION TITLE
Duplicates

### DISCUSSION THREAD

**Discussion by Baltimorepc** [LycheeOrg/Lychee#2848] [Q&A] [answered]
There was a thread where there was a issue  when uploading images it would cause duplicates. Has this been Resolved? if so how can i get rid of all the duplicates with out having to do it manually? As it created alot

**Comment by ildyria** [comment]
It has been resolved yes. However the duplication removal is a tricky one and may need some DB fiddling
(or wait for a duplicate searcher to be available in Lychee).

Could you check in your DB if a duplicate has the same sha thumbprint ? That would make it easier for me to create such page (rather than just trying to find on names...)

**Reply by Baltimorepc** [reply]
i am not sure, could you tell me how and where exactly to look, this spins up a seperate maria DB but the files are stored in a folder so i may be able to look there  i would be happy to check and let you  know.

**Reply by ildyria** [reply]
are you running on docker ?

**Reply by Baltimorepc** [reply]
Yes

**Reply by ildyria** [reply]
First quick question, when you have two duplicates, can you check if the medium image link is the same or different.
This will have some annoying implications if they are different...

**Reply by Baltimorepc** [reply]
checking now

**Reply by Baltimorepc** [reply]
the name is the same but im not sure where to look for the medium link. But i can tell you the actual link does show different. As if i scroll with t e arrow on the left when its full screen on the image. The name stays the same but the file is different in the website link. If thats what your asking

**Reply by ildyria** [reply]
right click on the picture, and check the link.
if for the same image name the link is different, this is going to be trouble some and hard to automate...

**Reply by ildyria** [reply]
I was hoping there would be an easy way out. But it seems that it will need either some duplicate finding engineering or creating the duplicate page finding earlier than I expected.

I'll give you a small SQL query to execute too, that should help with checking if there are duplicates. But I need to cook that later tomorrow. First sleep.

**Reply by Baltimorepc** [reply]
no worries, i appreciate the assistance. Thank you. the only real concern i have about the duplicate naming is when members upload to the same folder they maybe different pics but possibly the same name since they come from different device but same model. But i dont recall seeing any but truthfully i wasent really looking at the names unless i saw the same pic is when i actually glanced to see the name. So its possible there some but i could check if needed.

**Reply by ildyria** [reply]
We do have a checksum column in the DB which I plan to see if I can use in order to find duplicates (checksum never lie).

**Reply by ildyria** [reply]
Some useful queries for later...

Photos duplicates (checksum):
```sql
SELECT
    a.title as album_title,
    p.id as photo_id,
	p.title as photo_title,
    p.checksum
FROM
    photos as p
JOIN base_albums as a ON p.album_id = a.id
INNER JOIN photos as p2
ON p.checksum = p2.checksum AND p.id <> p2.id
ORDER BY p.checksum ASC;
```

Photos title duplicate in same album:
```sql
SELECT
    a.title as album_title,
    p.id as photo_id,
	p.title as photo_title,
    p.checksum
FROM
    photos as p
JOIN base_albums as a ON p.album_id = a.id
INNER JOIN photos as p2
ON p.title = p2.title AND p.id <> p2.id AND p.album_id = p2.album_id
ORDER BY a.title ASC, p.title ASC;
```

Pure photo duplicate in same album (title & checksum)
```sql
SELECT
    a.title as album_title,
    p.id as photo_id,
	p.title as photo_title,
    p.checksum
FROM
    photos as p
JOIN base_albums as a ON p.album_id = a.id
INNER JOIN photos as p2
ON p.title = p2.title AND p.checksum = p2.checksum AND p.id <> p2.id AND p.album_id = p2.album_id
ORDER BY a.title ASC, p.title ASC;
```

**Reply by ildyria** [reply]
That should make it easier for me to track those. :)

**Comment by ildyria** [comment, marked as answer]
Fix incomming: #2872

**Reply by Baltimorepc** [reply]
It looks like not only did it delete the duplicates it also added the normal file to that list and deleted them all not just the copies. I thought that list only shown duplicates so i deleted all that shown. and seems it deleted the first one too. so now yea i dont have dublicates but it removed the only copy with it.

**Reply by ildyria** [reply]
Well it show duplicates AND original. How am I supposed to know which one is the original ? :sweat_smile: 
So yes, if you tick everything, it deletes... everything.

**Reply by ildyria** [reply]
I thought that was clear.

**Reply by ildyria** [reply]
@Baltimorepc [`57fb224` (#2890)](https://github.com/LycheeOrg/Lychee/pull/2890/commits/57fb2246f818ec1cd7a5331847aa9f2cc7129818) added a warning :)
